test/e2e/main.grpc.test.ts

```diff
@@ -267,6 +270,94 @@ describe('grpc/main', () => {
+    it('should be able to get the configuration items with metadata', async () => {
+      // await server.configuration.subscribe();
+      await client.configuration.get("config-redis", ["myconfigkey1"], { "hello": "world" });
+
+      // Disabled for now as I am unsure if Dapr returns the metadata items
+      // Java SDK: https://github.com/dapr/java-sdk/blob/06d92dafca62a6b48e74ccf939feeac7189e360f/sdk/src/test/java/io/dapr/client/DaprPreviewClientGrpcTest.java#L119
+      // ^ shows that it is not being tested, it tries but doesn't assert
+      // expect(conf.items.filter(i => i.key == "myconfigkey1")[0].metadata).toHaveProperty("hello");
+    });
+
+    // // @todo: figure out, subscribe doesn't pass keys but doesn't listen to anything?
+    // // https://github.com/dapr/dapr/issues/4529
+    // it('should be able to subscribe to configuration item changes on all keys', async () => {
+    //   const m = jest.fn(async (res: SubscribeConfigurationResponse) => { });
+
+    //   console.log("Creating Subscription");
+    //   await client.configuration.subscribe("config-redis", m);
+
+    //   // Change an item
+    //   await DockerUtils.executeDockerCommand("dapr_redis redis-cli MSET myconfigkey3 mynewvalue||1");
+
+    //   expect(m.mock.calls.length).toEqual(1);
+    // });
+
+    it('should be able to subscribe to configuration item changes on specific keys', async () => {
+      const m = jest.fn(async (_res: SubscribeConfigurationResponse) => { return; });
+
+      await client.configuration.subscribeWithKeys("config-redis", ["myconfigkey1", "myconfigkey2"], m);
+      await DockerUtils.executeDockerCommand("dapr_redis redis-cli MSET myconfigkey1 mynewvalue||1");
+
+      expect(m.mock.calls.length).toEqual(1);
+      expect(m.mock.calls[0][0].items[0].key).toEqual("myconfigkey1");
+      expect(m.mock.calls[0][0].items[0].value).toEqual("mynewvalue");
+    });
+
+    it('should be able to subscribe to configuration items through multiple streams', async () => {
+      const m1 = jest.fn(async (_res: SubscribeConfigurationResponse) => { return; });
+      const m2 = jest.fn(async (_res: SubscribeConfigurationResponse) => { return; });
+
+      await client.configuration.subscribeWithKeys("config-redis", ["myconfigkey1"], m1);
+      await client.configuration.subscribeWithKeys("config-redis", ["myconfigkey1"], m2);
+
+      await DockerUtils.executeDockerCommand("dapr_redis redis-cli MSET myconfigkey1 mynewvalue||1");
+
+      expect(m1.mock.calls.length).toEqual(1);
+      expect(m1.mock.calls[0][0].items[0].key).toEqual("myconfigkey1");
+      expect(m1.mock.calls[0][0].items[0].value).toEqual("mynewvalue");
+
+      expect(m2.mock.calls.length).toEqual(1);
+      expect(m2.mock.calls[0][0].items[0].key).toEqual("myconfigkey1");
+      expect(m2.mock.calls[0][0].items[0].value).toEqual("mynewvalue");
+    });
+
+    it('should be able to subscribe to configuration item changes on specific keys through a second stream', async () => {
```

Comment:
What is the difference between this test and "should be able to subscribe to configuration item changes on specific keys"?

Reply:
It ensures we can open 2 different streams to the same config key

Reply:
Sorry, where is the second stream being opened? Are you referring to another test case?

Reply:
Removed a test case